### ISSUE TITLE
[Feature] Geo operations over ArangoSearch/View

### ISSUE THREAD

**Issue by aveiros** [arangodb/arangodb#10566]
My understanding is that Geo Operations have to fall back into FILTER, which makes queries large data sets extremely slow.

Another limitation (might have been slowed now), is that Geo Indexes to not support structures as follow:

```
{
   "geoPoints": [
         { "lat": 1, "lon": 2 },
         { "lat": 5, "lon": 6 }
    ]
}
```

The proposal would be to add support for Geo Operations on Arango Search (SEARCH), thus combining full-text search with geo operations.

On the View definition, pattern matching on Geo Structure Fields would be ideal.

Use cases:
Find "sushi" and "restaurants" within lat,lon and radius 500 meters.

**Comment by graetzer** [comment]
I don't get it you can easily use multipoint GeoJSON values

**Comment by aveiros** [comment]
@graetzer 

In the ideal scenario, yes. But in my case, working with a legacy application (legacy model).

Making it part of SEARCH and providing the ability to configure the geo fields addresses both worlds.

**Comment by admtech** [comment]
Here an Example: We do a Jobsearch (DB jobs) with Distance (40 KM) and Term Search (Windows Administrator). "coordinates" are a GEO JSON Field. We have a view (job_search) and a Geo Index

```
LET query = "Windows Administrator"
LET myLocation = GEO_POINT(8.168013, 50.364549)
FOR doc IN job_search
    SEARCH
    ANALYZER(
      BOOST(doc.title IN TOKENS(query, "text_de"),2) OR
      BOOST(STARTS_WITH(doc.title, query),2) or
      PHRASE(doc.desc, query)
    ,"text_de")
    FILTER GEO_DISTANCE(myLocation, doc.job_location.coordinates) <= 40000
    SORT GEO_DISTANCE(myLocation, doc.job_location.coordinates) ASC
    LIMIT 50
    RETURN {doc}
```
Result: Uncached: 19.369 ms, Cached: 4.325 ms
Without Filter and Sort (GEO_DISTANCE): Uncached: 1.225 ms, Cached: 0.845 ms
That's a big difference.

I also tried a join, then it took the GEO index, but the whole thing was even slower (>90 ms). Is there another way or join examples that work?

best regards
Frank

**Comment by gnusi** [comment]
@aveiros this is addressed by ArangoSearch GEO functionality in 3.8

**Comment by aveiros** [comment]
Thanks @gnusi & the team.
